src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -133,27 +133,46 @@ const ContextMenu = createClass({
 	},
 
 	componentDidMount() {
-
 		_.defer(() => this.alignFlyOut());
 		this.updateTargetRectangleIntervalId = setInterval(() => {
 			if (this.props.isExpanded) {
 				this.alignFlyOut();
 			}
 		}, 10);
 
-		this.onClickBodyEventListener = window.addEventListener('click', (event) => {
-			if (this.props.onClickOut && this.refs.flyOutPortal) {
-				const flyOutEl = this.refs.flyOutPortal.portalElement.firstChild;
-				if (!(flyOutEl.contains(event.target) || this.refs.target.contains(event.target))) {
-					this.props.onClickOut({ props: this.props, event });
-				}
-			}
-		});
+		document.body.addEventListener('click', this.handleBodyClick);
 	},
 
 	componentWillUnmount() {
 		clearInterval(this.updateTargetRectangleIntervalId);
-		document.body.removeEventListener('click', this.onClickBodyEventListener);
+		document.body.removeEventListener('click', this.handleBodyClick);
+	},
+
+	handleBodyClick(event) {
+		const {
+			props,
+			props: {
+				onClickOut,
+			},
+			refs: {
+				flyOutPortal,
+				target,
+			},
+		} = this;
+
+		if (onClickOut && flyOutPortal) {
+			const flyOutEl = flyOutPortal.portalElement.firstChild;
+			const wasALabelClick = event.target.nodeName === 'INPUT' && sharesAncestor(event.target, target, 'LABEL');
```

Comment:
I learned that `<label>`s fire a real click event with a target of the corresponding `INPUT`. This line checks to see if the event fired relates to an input and shares a label ancestor with the context menu target.